### ISSUE TITLE
Missing docs warning from linter when using `#[bon]` macro

### ISSUE THREAD

**Issue by matthewgrimes** [elastio/bon#38]
The following code
```
#![warn(missing_docs)]
use bon::bon;                                                                                                                                                                 
  
struct MyStruct;
  
#[bon]
impl MyStruct {
    #[builder]
    pub fn new() -> Self {
        Self
    }
}
fn main() {
      println!("Hello, world!");
 }
```

raises linter warnings
```
warning: missing documentation for a struct
 --> src/main.rs:6:1
  |
6 | #[bon]
  | ^^^^^^
  |
  = note: this warning originates in the attribute macro `bon` (in Nightly builds, run with -Z macro-backtrace for more info)
```

I'm using v1.0.6 of `bon` and `rustc 1.80.0`.

**Comment by Veetaha** [comment]
Hi, thank you for opening the issue! I'll fix this in the upcoming 1.1 minor release. It should be available some time tomorrow

**Comment by Veetaha** [comment]
Btw, in the example you shown there are no docs for the `new` method, that would be a valid missing docs lint. The problem that the builder struct has no docs is fixable inside bon though

**Comment by matthewgrimes** [comment]
Thanks, I appreciate it!

>Btw, in the example you shown there are no docs for the new method, that would be a valid missing docs lint.

Agreed, there will be multiple linter warnings in my example but just wanted to get enough to trigger the `#[bon]` warning.

**Comment by Veetaha** [comment]
The fix for this particular example of code in the issue was merged in #38 and it should be available starting with the version `1.1.0` of `bon`. Note that the fix included adding docs to the generated builder struct only. You'll still get warnings about missing docs if your function's arguments or struct's fields aren't documented. (See ["documenting" guide](https://elastio.github.io/bon/docs/guide/documenting) for details about documenting function's arguments).

This may be a bit confusing that you need to add docs to function's arguments, but these docs are the ones that will be inherited by the setter methods. I'm not yet sure of `bon` should add an `allow` for this lint on setter methods. You'll get warnings about missing docs on setter methods.

Maybe setters should have some default dumb documentation generated if there are no docs. E.g. smth like `Set the field/parameter {foo}`

**Comment by matthewgrimes** [comment]
I think a default documentation when one isn't provided makes sense. Even `Set {foo}` seems adequate. Or maybe something to indicate the method is auto-generated? `This method was auto-generated using bon -- see {function declaration}/{struct impl block} for more documentation`?

**Comment by Veetaha** [comment]
Yeah makes sense, I'll reopen the issue to do this

**Comment by Veetaha** [comment]
With the PR #45, the docs are now also autogenerated on setter methods. This change will be released as part of the `1.2` minor release which also added the `#[builder(skip)]` attribute (#44).

![image](https://github.com/user-attachments/assets/618b72f6-5af7-44ee-b653-05e601759845)

**Comment by matthewgrimes** [comment]
Thank you!
